### PR TITLE
New release: Python3 + Cloud + Responsiv + etc

### PR DESCRIPTION
**Pull request by the-maux**
* Migrate Python2.7 to Python3.7 and upgrading all python library
    * Removed libraries:
        * mock==1.0.1
        * requests==1.2.3
        * six>=1.4.1
        * tornado-redis==2.4.2
        * tornado==3.0.1
        * docopt==0.6.1
        * raven==3.5.2
        * Sphinx==1.2.2
        * sphinxcontrib-httpdomain==1.3.0

    * Added libraries:
        * django>=2.2.1
        * django-prometheus>=1.0.15
        * django-cors-headers>=3.0.1
        * channels>=2.2.0
        * channels_redis>=2.4.0
        * asgiref>=3.1.4

* Using Django2.0 over tornado now
    * Updating tornado template tiles-.html to template django, moving them in templates directory
    * Using channels_redis.core.RedisChannelLayer over tornadoredis
    * Using channels.generic.websocket over tornado.websocket.WebSocketHandler
    * Using static files from django behavior over tornado.web.StaticFileHandler
    * Moving tornado url routing behavior to the django behavior
    * removing duplicate inner function to a single file utils.py
    * reworking of the file redis_utils.py to a cache.py
    * Config for ASGI_APPLICATION

* Moving the .tipboard directory to ./tipboard/Config/ to make it compatible with bitnami/python
* Adding dockerfile support
    * Using bitnami image in order to be cloud ready
    * Using non root image
* Adding .gitlab-ci to CI with gitlab
* Minor fixes, improvements, cleanups etc.

### CODE REVIEW

**Comment by the-maux** [comment]
Update:
* Fixed issue who caused latency with websocket on first load
* Added CI on docker hub here : https://hub.docker.com/r/themaux/tipboard
* Added github package https://github.com/the-maux/tipboard/packages/14519
* Update Readme to show quick usage example (w.i.p)
* Update unit-test (w.i.p)

**Comment by the-maux** [comment]
Update:
https://github.com/the-maux/tipboard/commit/20db63eae1452fdff2ae6af4969540204e20a3d8 to answer at issue #21 
features(api): meta & update tile on same request /update
For example, before, your data for the request was:
```json  
  var = {"tile": "big_value",
           "key":  "mybigvalue",
           "data": {"title": "% of something",
                               "description": "% of subtitle",
                               "big-value": "50%",
                               "lower-left-label": "title_value1:",
                               "upper-left-label": "title_value2:",
                               "upper-right-label": "value1",
                               "lower-right-label": "value2"}
         }
```
It's now:
```json  
  var = {"tile": "big_value",
           "key":  "mybigvalue",
           "value": {"big_value_color": "green", "fading_background": "true"},
           "data": {"title": "% of something",
                               "description": "% of subtitle",
                               "big-value": "50%",
                               "lower-left-label": "title_value1:",
                               "upper-left-label": "title_value2:",
                               "upper-right-label": "value1",
                               "lower-right-label": "value2"}
           }
```
But you can still use /push & /tileconfig

**Comment by jbarbera** [comment]
I m still having the "" name 'execfile' is not defined "" issue trying to create an empty project, am i missing something? working with python 3.6

**Comment by the-maux** [comment]
What command do you use ?
If you use pip, the package tipboard was not update.
You don't need to create an empty project now, by source just type:

> $> python manage runserver 

Python 3.7 is required now

**Comment by jbarbera** [comment]
cool , thanks i ll give it a try :D

**Comment by the-maux** [comment]
Hey @jbarbera
After 7ac7002 you can now install with pip
```shell
$> virtualenv -p python3 tb-env
$> source tb-env/bin/activate
(tb-env)$> pip install tipboard2.0
(tb-env)$> tipboard runserver 0.0.0.0:8080
```
**It's in WIP** to modify the layout_config.yml, but the package but will be automaticaly update on pypi when fixed, thanks to the [pipeline circle-ci](https://circleci.com/gh/the-maux/tipboard/38?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)   
Status Badge: [CircleCI](https://circleci.com/gh/the-maux/tipboard/tree/develop)

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,54 @@
 Change Log
 ----------
 
+1.5.0
```
Comment (fatalcoder) [inline CHANGES.rst:50]: readme is for version 1.5 but `src/__init__.py` `__version__` says 2.0.0

Comment (the-maux) [reply CHANGES.rst:50]: Updated with 335d1c1

**Review by the-maux — COMMENTED** [review]
first check

**Comment by the-maux** [comment]
Closing the PR

**Comment by the-maux** [comment]
**679 commit later**, i'm back and almost ready to publish Tipboard 3.0,
I will start a new depot, named Chartboard under M.I.T License (except if we merge this PR :smiley:).
I will soon archived my working depot, to start a full new project with a new name (like piwik for matomo :laughing:)
Asap, I will put a full list of changes in the Change.rst and here, also[ the migration story](https://github.com/the-maux/tipboard/wiki/Tipboard-migration) in the wiki
Don't hesitate to check the new [wiki](https://github.com/the-maux/tipboard/wiki)
I hope you will enjoy it, there are a lot of new features :)

**Comment by vi4m** [comment]
Thanks for keeping working on this fork, looks really promising.

**Comment by the-maux** [comment]
@vi4m thanks 😄 
Don't hesitate to check the new fork https://github.com/the-maux/Chartboard 
Bc Covid, i couldn't work like i wanted on it since, but it's almost production ready 
I think i need help on documentation, **i'm so fucking lazy about it** !
But i will do it, Chartboard v1.0 wil be, and maybe one day we will full merge ❤️

**Comment by vi4m** [comment]
How's the work going? Is it ready?

**Comment by the-maux** [comment]
I didn't saw your message sorry. 
But yes, Check it out :https://github.com/the-maux/Chartboard

**Pull request closed without merging**